Fuzzlyn/Types/PrimitiveType.cs

```diff
@@ -271,27 +271,30 @@ private static T FromTypicalLiteralLoopLowerBound<T>(TypicalLiteralKind literal)
             TypicalLiteralKind.Two => T.CreateChecked(2),
             TypicalLiteralKind.Ten => T.CreateChecked(10),
             TypicalLiteralKind.MaxValueMinusOne => T.MaxValue - T.CreateChecked(100), // This is a bit misnamed for loop bounds
-            TypicalLiteralKind.MaxValue => T.MaxValue - T.CreateChecked(5), // This is a bit misnamed for loop bounds, since we need space after it
+            TypicalLiteralKind.MaxValue => T.MaxValue - T.CreateChecked(50), // This is a bit misnamed for loop bounds, since we need space after it
             _ => throw new ArgumentOutOfRangeException(nameof(literal)),
         };
     }
 
-    // TODO: handle varying loop iteration counts
-    private static T FromTypicalLiteralLoopUpperBound<T>(TypicalLiteralKind literal) where T : INumber<T>, IMinMaxValue<T>
+    private static T FromTypicalLiteralLoopUpperBound<T>(TypicalLiteralKind literal, Randomizer random) where T : INumber<T>, IMinMaxValue<T>
     {
+        // An iteration count of 2 leads to JIT loop unrolling (if size contraints apply). An iteration count of 40 is larger
+        // than the JIT allowed constant loop iteration unroll count, and leads to loop cloning.
+        T iterationCount = random.FlipCoin(random.Options.SmallLoopIterationCountProb) ? T.CreateChecked(2) : T.CreateChecked(40);
```

Comment:
With the high probability of generating 40 iterations I would be worried about generating many nested loops of with 40 iterations. This probably should have some kind of budget inside `GenLoop`.

The nesting level is controlled by this hill equation: https://www.desmos.com/calculator/oiftyfbfrh
It makes the nesting of a bunch of for loops rare, but since the curve is O(40^n) it grows very rapidly.

Reply:
I was thinking an alternative would be to keep the iteration count low, but put the iteration end in a global static variable, so maybe the JIT wouldn't be able to "inline" the value.

Reply:
> It does pollute all generated loops with an early exit, making the lower bound on the trip count unknowable, which excludes a whole class of possible loop transformations.

I think that makes this option a non-starter.

Reply:
I think all RyuJITs loop transformations handle multiple exits (in practice, due to exceptions, almost all loops come with unknowable lower bounds on the trip count). So I think this budget based approach would be much better for coverage than putting the iteration end in a static variable, where none of the JIT's IV opts would be able to reason about it.